### PR TITLE
types: migrates test-infra to testify for `helper_test.go`

### LINKED ISSUE #28007: migrate test-infra to testify for `helper_test.go`

### PR DESCRIPTION
**Pull request by dswij**
Close #28007

What's Changed: Test in format_test.go now uses `testify/require` instead

```release-note
None
```

### CODE REVIEW

File: types/helper_test.go [hunk 1]
```diff
@@ -42,13 +39,14 @@ func (s *testTypeHelperSuite) TestStrToInt(c *C) {
 	}
 	for _, tt := range tests {
 		output, err := strToInt(tt.input)
```
Comment (feitian124) [inline types/helper_test.go:41]: please make sure range and t.Parallel works right.
see https://github.com/pingcap/tidb/pull/27939 for deail

**Comment by feitian124** [comment]
LGTM
please check if ci failed due to https://github.com/pingcap/tidb/issues/25899

**Review by mmyj — APPROVED** [review]
lgtm

**Review by tisonkun — APPROVED** [review]
LGTM. Thanks for your contribution @dswij !

**Pull request merged**